File: UI/MyInterviewPreparation/ClientApp/src/services/questionanswer.service.ts
import axios from "axios";
import { UserInfo } from "../Entities/UserInfo";
import { UserQuestionAnswer } from "../Entities/UserQuestionAnswer";
import { UserSubject } from "../Entities/UserSubject";

const API_URL = "https://localhost:4579/api/QuestionAnswer/";

export const GetAllUserDomainBySubjectId = (subjectId: number) => {
  return axios
    .get(API_URL + "GetAllUserDomainBySubjectId", {
      params: {
        subjectId,
      }
    })
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const GetAllUserQuestionAnswerByUserId = (userId: number) => {
  return axios
    .get(API_URL + "GetAllUserQuestionAnswerByUserId", {
      params: {
        userId,
      }
    })
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const GetAllUserQuestionAnswerByDomainId = (domainId: number) => {
  return axios
    .get(API_URL + "GetAllUserQuestionAnswerByDomainId", {
      params: {
        domainId,
      }
    })
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const GetAllUserSubjectByUserId = (userId: number) => {
  return axios
    .get(API_URL + "GetAllUserSubjectByUserId", {
      params: {
        userId,
      }
    })
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const GetUserQuestionAnswerById = (id: number) => {
  return axios
    .get(API_URL + "GetUserQuestionAnswerById", {
      params: {
        id,
      }
    })
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const GetRandomQuestionAnswerByDomainId = (domainId: number) => {
  return axios
    .get(API_URL + "GetRandomQuestionAnswerByDomainId", {
      params: {
        domainId,
      }
    })
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};


export const SaveUserQuestionAnswerById = (userQuestionAnswer: UserQuestionAnswer) => {
  return axios
    .post(API_URL + "SaveUserQuestionAnswerById", 
      userQuestionAnswer,
    )
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const SaveUserSubject = (userSubject: UserSubject) => {
  return axios
    .post(API_URL + "SaveUserSubject", 
      userSubject,
    )
    .then((response: any) => {
      return JSON.parse(response.data.data).payload;
    });
};

export const register = (email: string, password: string) => {

  return axios.post(API_URL + "RegisterUser", {
    BaseURL: "abc",
    MethodName: "abc",
    email,
    password,

  }).then((response: any) => {
    console.log(response);

    if (response.data) {
      localStorage.setItem("user", JSON.stringify(response.data));
      localStorage.setItem("accessToken", JSON.stringify(response.accessToken));
    }

    return JSON.parse(response.data.data).payload;
  });
};



export const logout = () => {
  localStorage.removeItem("user");
  window.location.href = "login";
};

export const getCurrentUser = () => {
  const userStr = localStorage.getItem("user");
  if (userStr) return JSON.parse(userStr)?.payload?.userInfo as UserInfo;

  return null;
};

File: Services/QuestionAnswer.API/Service/QuestionAnswerService.cs
using CommonEntities.Auth;
using CommonEntities.QuestionAnswer;
using CommonEntities.Result;
using QuestionAnswer.API.Repository;
using System.Collections.Generic;

namespace QuestionAnswer.API.Service
{
    public class QuestionAnswerService : IQuestionAnswerService
    {
        public IQuestionAnswerRepository _questionAnswerRepository;
        public QuestionAnswerService(IQuestionAnswerRepository questionAnswerRepository)
        {
            _questionAnswerRepository = questionAnswerRepository;
        }
        public async Task<ServiceResultWithData<IEnumerable<UserDomain>>> GetAllUserDomainBySubjectId(int subjectId)
        {
            var result = new ServiceResultWithData<IEnumerable<UserDomain>>();
            result.Data = await _questionAnswerRepository.GetAllUserDomainBySubjectId(subjectId);
            return result;
        }

        public async Task<ServiceResultWithData<IEnumerable<UserQuestionAnswer>>> GetAllUserQuestionAnswerByDomainId(int domainId)
        {
            var result = new ServiceResultWithData<IEnumerable<UserQuestionAnswer>>();
            result.Data = await _questionAnswerRepository.GetAllUserQuestionAnswerByDomainId(domainId);
            return result;
        }

        public async Task<ServiceResultWithData<IEnumerable<UserQuestionAnswer>>> GetAllUserQuestionAnswerByUserId(int userId)
        {
            var result = new ServiceResultWithData<IEnumerable<UserQuestionAnswer>>();
            result.Data = await _questionAnswerRepository.GetAllUserQuestionAnswerByUserId(userId);
            return result;
        }

        public async Task<ServiceResultWithData<IEnumerable<UserSubject>>> GetAllUserSubjectByUserId(int userId)
        {
            var result = new ServiceResultWithData<IEnumerable<UserSubject>>();
            result.Data = await _questionAnswerRepository.GetAllUserSubjectByUserId(userId);
            return result;
        }

        public async Task<ServiceResultWithData<UserQuestionAnswer>> GetRandomQuestionAnswerByDomainId(int domainId)
        {
            var result = new ServiceResultWithData<UserQuestionAnswer>();
            result.Data = await _questionAnswerRepository.GetRandomQuestionAnswerByDomainId(domainId);
            return result;
        }

        public async Task<ServiceResultWithData<UserQuestionAnswer>> GetUserQuestionAnswerById(int id)
        {
            var result = new ServiceResultWithData<UserQuestionAnswer>();
            result.Data = await _questionAnswerRepository.GetUserQuestionAnswerById(id);
            return result;
        }

        public async Task<ServiceResultWithData<bool>> SaveUserQuestionAnswer(UserQuestionAnswer userQuestionAnswer)
        {
            var result = new ServiceResultWithData<bool>();
            result.Data = await _questionAnswerRepository.SaveUserQuestionAnswer(userQuestionAnswer);
            return result;
        }

        public async Task<ServiceResultWithData<bool>> SaveUserQuestionAnswerById(UserQuestionAnswer userQuestionAnswer)
        {
            var result = new ServiceResultWithData<bool>();
            result.Data = await _questionAnswerRepository.SaveUserQuestionAnswerById(userQuestionAnswer);
            return result;
        }

        public async Task<ServiceResultWithData<bool>> SaveUserSetting(UserSetting userSetting)
        {
            var result = new ServiceResultWithData<bool>();
            result.Data = await _questionAnswerRepository.SaveUserSetting(userSetting);
            return result;
        }

        public async Task<ServiceResultWithData<bool>> SaveUserSubject(UserSubject userSubject)
        {
            var result = new ServiceResultWithData<bool>();
            result.Data = await _questionAnswerRepository.SaveUserSubject(userSubject);
            return result;
        }

        public async Task<ServiceResultWithData<bool>> UpdatePrepairdRatioByQuesionAnswerId(int QuesionAnswerId)
        {
            var result = new ServiceResultWithData<bool>();
            result.Data = await _questionAnswerRepository.UpdatePrepairdRatioByQuesionAnswerId(QuesionAnswerId);
            return result;
        }
    }
}
